### PR TITLE
[SG-27703] Wildcard: `<NavMenu />` Implementation

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description
We already have `<Menu />`, `<MenuDivider />` accessible components here: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@tr/wildcard-popover/-/blob/client/wildcard/src/components/Menu/Menu.tsx#L1:1
We want to be able to easily implement navigation menus

## Test Plan
- Visit the app storybook preview link
- Confirm the the `NavMenu` story matches desired result.
- Confirm that Component has all required APIs that are needed by consumer.

## Ref 
[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/27703)
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-27703)

## Preview App
[Web App](https://client-sourcegraph-frontend-pr-286.onrender.com/)
[Storybook](https://client-sourcegraph-storybook-pr-286.onrender.com/)

Acceptance criteria
 - [x] Component is implemented within Wildcard
 - [x] Component has tests
 - [x] Component is accessible
 - [x] Component matches [Figma designs](https://www.figma.com/file/NIsN34NH7lPu04olBzddTw/Wildcard-Design-System?node-id=908%3A1)

## Time Estimate
10hrs - 12hrs
## App preview:
- [Link](https://sg-web-contractors-sg-27703.onrender.com)

### CODE REVIEW

File: client/wildcard/src/components/NavMenu/NavMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        {sections.map((sections, index) => (
+            <NavMenuSection {...sections} hideDivider={index !== 0} key={index} />
+        ))}
+    </MenuList>
+)) as ForwardReferenceComponent<'div', NavMenuContentProps>
+
+interface NavItemProps extends Omit<MenuItemProps, 'children' | 'onSelect'> {
+    prefixIcon?: React.ComponentType<MdiReactIconProps>
+    suffixIcon?: React.ComponentType<MdiReactIconProps>
+    itemContent: string | ReactNode
```
Comment (oleggromov) [inline client/wildcard/src/components/NavMenu/NavMenu.tsx:59]: Let's rename this to `content` as `item` is implied from `navItems` key in `NavMenuSectionProps`.

File: client/wildcard/src/components/NavMenu/NavMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    if (to) {
+        return (
+            <MenuLink as={itemAs || Link} to={to} {...rest}>
+                {content}
+            </MenuLink>
+        )
+    }
+
+    return (
+        <MenuText as={itemAs || MenuItem} {...rest}>
```
Comment (oleggromov) [inline client/wildcard/src/components/NavMenu/NavMenu.tsx:91]: I tried to make the type `NavItemProps.itemAs` strict `itemAs?: React.Component` as we do in other Wildcard components and ran into multiple typing issues.

After tinkering with typing, I am wondering if there's any use case for redefining the way `MenuItem`, `MenuLink`, or `MenuText` are rendered? Except the theme menu section, which uses `itemAs` but it doesn't change anything.

Maybe it's easier to simply remove the `itemAs` property for now, as it's not used at all?

**Review by oleggromov — CHANGES_REQUESTED** [review]
Hey, thanks for this contribution!
I left a few code-related comments, please have a look.

And another big concern, even though it's the same in prod now, when I use keyboard arrow keys to navigate among menu items, the theme selectbox never gets selected for some reason.

Any ideas why it is like that?

https://user-images.githubusercontent.com/2196347/160813341-3fd5a219-6083-432a-b605-2983c5988da7.mov

**Comment by gitstart-sourcegraph** [comment]
> And another big concern, even though it's the same in prod now, when I use keyboard arrow keys to navigate among menu items, the theme selectbox never gets selected for some reason.

The reason this happens in prod version, is that the theme selection row, is not wrapped around `MenuItem` component, as is the case for other nav items. In current implementation (in storybook), we attempted to replicate the exact behaviour of the existing system.

I believe why it was not wrapped in a `MenuItem` is that, when the theme Select input is clicked, the MenuItems's `onSelect` callback would be triggered, which by default closes the `Menu` after it's execution. Since `MenuItem` does not provide the event in the `onSelect` callback, there would be no way to `preventDefault`
[Ref here](https://github.com/reach/reach-ui/blob/develop/packages/dropdown/src/index.tsx#L351-L355)
@oleggromov

**Review by oleggromov — APPROVED** [review]

**Pull request merged**